ding_dibs.module

```diff
@@ -420,3 +442,142 @@ function ding_dibs_cards() {
 
   return $cards;
 }
+
+/**
+ * Implements hook_cron().
+ * This is done by sending written reports and email adresses result cron launches in the database
+ */
+function ding_dibs_cron() {
+  $start_cron = FALSE;
+  $start = variable_get('ding_dibs_start_cron', '');
+  $start_minute = variable_get('ding_dibs_start_cron_minute', 0);
+  $start_operation = variable_get('ding_dibs_start_operation_cron', '');
+  $emails_list = variable_get('ding_dibs_emails', '');
+  if (!empty($start)) {
+    $start_val = mktime($start, $start_minute, 0, format_date(REQUEST_TIME, 'custom', 'm'), format_date(REQUEST_TIME, 'custom', 'd'), format_date(REQUEST_TIME, 'custom', 'Y'));
+    $today_date = format_date(REQUEST_TIME, 'custom', 'd/m/Y/H:i');
+    $start_date = format_date($start_val, 'custom', 'd/m/Y/H:i');
+    if (!empty($start_operation)) {
+      $start_operation_date = format_date($start_operation, 'custom', 'd/m/Y');
+    }
+    if ($today_date == $start_date || $today_date >= $start_date) {
+      if (empty($start_operation_date) || $start_operation_date < format_date(REQUEST_TIME, 'custom', 'd/m/Y')) {
+        $start_cron = TRUE;
+      }
+    }
+  }
```

Comment:
The code in this function up until this line seems to be a very convoluted way of determining if the reports should be generated and sent.

The use of comparison and strings Lines 463 and 464 also makes we wonder whether it works across months. For example `'31/01/15' > '01/02/15'` evaluates to `true`.

This needs to be rewritten to be more sol. I encourage you to take a look at the [PHP Datetime API](http://php.net/manual/en/book.datetime.php).